sm/engine/dataset_manager.py

```diff
@@ -170,19 +177,19 @@ def delete(self, ds, del_raw_data=False, **kwargs):
 
 class SMapiDatasetManager(DatasetManager):
 
-    def __init__(self, qname, db, es, mode, queue_publisher=None):
+    def __init__(self, qname, db, es, image_store, mode, queue_publisher=None):
         self.qname = qname
-        DatasetManager.__init__(self, db=db, es=es, mode=mode,
+        DatasetManager.__init__(self, db=db, es=es, img_store=image_store, mode=mode,
                                 queue_publisher=queue_publisher, logger_name='sm-api')
 
     def _post_sm_msg(self, ds, action, priority=DatasetActionPriority.DEFAULT, **kwargs):
+        ds.set_status(self._db, self._es, self._queue, DatasetStatus.QUEUED)
```

Comment:
@iprotsyuk Why to move it here?

Reply:
somehow in VBox installtion a published message is sometimes delivered before the dataset status is set. In this case, setting the status also saves the dataset in DB. So, if a message is processed before that moment, a message handler doesn't find the dataset in DB and the processing fails